xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp

```diff
@@ -123,14 +128,44 @@ bool CVAAPIContext::EnsureContext(CVAAPIContext **ctx, CDecoder *decoder)
   return true;
 }
 
+namespace
+{
+int find_open_render_node_fd()
```

Comment:
If you finally want to display a decoded frame on a monitor, you have a dependency to that system, whatever it is. Consider a multi-gfx-card, multi-monitor system. That means that windowing has to be involved in selecting the right render node. It is NOT sufficient to select any render node for the give use case.